pkg/common/clock/clock.go

```diff
@@ -0,0 +1,120 @@
+package clock
+
+import (
+	"testing"
```

Comment:
Feels a little unsanitary for the testing package to end up in production code. Is it worth moving the mock stuff into `test/mocks/mockclock` or `test/fakes/fakeclock` or something?